### ISSUE TITLE
Redirect loop error on Survey Plugin Admin

### ISSUE THREAD

**Issue by deetweet** [drohne1673/confluence-survey-plugin#69]
Hi,

I have an issue with the add-on, whenever a user (even ones in the Confluence Administrators group) goes to Confluence Admin and clicks on the 'Survey Plugin Administration' link, the browser times out with a 'This webpage has a redirect loop'. The full error message is:

The webpage at https://myconfluence/login.action?os_destination=https%3A%2F%2Fmyconfluence%2Fplugins%2Fservlet%2Fsurveyplugin%2Fadmin has resulted in too many redirects. Clearing your cookies for this site or allowing third-party cookies may fix the problem. If not, it is possibly a server configuration issue and not a problem with your computer.
Learn more about this problem.
Error code: ERR_TOO_MANY_REDIRECTS

I've checked the browser settings, cleared my cookies and made sure third party cookies are allowed.

Would appreciate if I could get some help with this error.

Thanks, David

**Comment by deetweet** [comment]
Just to add to the above:
* Confluence version 5.1
* Latest version of the add-on installed.
* I've tried the re-install with the same issue.
* I can spin up a test confluence instance but cannot replicate the issue.
* The production Confluence server with this issue is behind a HAProxy service (http://www.haproxy.org/).

**Comment by drohne1673** [comment]
Hy David,

confluence is strongly coupled with the base url you have configured. The functionality is quite straight forward and mostly taken from the confluence recommended tutorial: https://github.com/drohne1673/confluence-survey-plugin/blob/master/src/main/java/org/hivesoft/confluence/admin/AdminServlet.java

If the user is not logged in, do a redirect to the login screen.

What i dont understand, `even ones in the administrator group` everyone in the admin section must have that group, dont they?

But i would guess that you maybe are connecting from another base URL to the administration section than your base url is. 

The sample `https://myconfluence/login.action?os_destination=https%3A%2F%2Fmyconfluence%2Fplugins%2Fservlet%2Fsurveyplugin%2Fadmin` looks quite straight forward, can you check if that link works by just pasting it in the url?

thanks, Frank

**Comment by deetweet** [comment]
Hi Frank,

Thanks for the reply!

* Sorry I meant to say that the users having this redirect problem with the survey plugin are confluence administrators. A member of the system administrator group doesn't have this issue.
* I've tried just pasting the URL into the address bar and that doesn't work either, I still get the 'too many redirects' error after a few seconds.
* That URL is what is generated when I click on 'Survey Plugin Administration' on the Confluence admin screen.
* The same redirect happens whether I'm logged in to confluence or not.
-- If I'm logged in the error appears after a few seconds.
-- If I'm not logged in, I get the login screen, login successfully but then get the error after a few seconds.
* I haven't tried restarting confluence to see if that resolves the issue, I have re-installed the add-on with no change.

**Comment by drohne1673** [comment]
i think we can verify the behavior quite comfortable with the given URI. If you say that it works not really well. The question is, whether anyother destination will work. The login.action is the central login into confluence, and after the user has been logged in the browser gets redirected to the given destination.

can you check that this works generally, simply by invoking something like `https://myconfluence/login.action?os_destination=https%3A%2F%2Fmyconfluence%2Fsomespace%2Fsomepage`

**Comment by deetweet** [comment]
Yes, I can confirm that all other add-ons and pages work fine. This is only affecting the survey configuration link.

**Comment by drohne1673** [comment]
This works  (plugin administration):
`http://localhost:8090/login.action?os_destination=http%3A%2F%2Flocalhost%3A8090%2Fplugins%2Fservlet%2Fsurveyplugin%2Fadmin`

This works (regular page login):
`http://localhost:8090/login.action?os_destination=%2Fpages%2Fviewpage.action%3Ftitle%3DYet%2Banother%2BSurvey%2Btest%26spaceKey%3DSRV&permissionViolation=true`

Your sample (same as first / plugin administration), only difference https:
`https://myconfluence/login.action?os_destination=https%3A%2F%2Fmyconfluence%2Fplugins%2Fservlet%2Fsurveyplugin%2Fadmin`

> If I'm not logged in, I get the login screen, login successfully but then get the error after a few seconds.
can you elaborate on ... after a few seconds ... so can you actually see the admin section of the survey plugin and then after a few seconds something happens? But what, without interaction, there is no timer running ;)

**Comment by drohne1673** [comment]
Ok i checked some other plugins. Lets try the relative login, can you try that with your base url:

`http://localhost:8090/login.action?os_destination=%2Fplugins%2Fservlet%2Fsurveyplugin%2Fadmin`

That is the relative uri for the admin section instead of the absolute one, that worked in all my plugins

**Comment by deetweet** [comment]
Hi Frank,

Thanks for following up. So, I'm referencing your examples as listed but I changed the link to be valid in my environment. In all cases:

http://localhost:8090/login.action?os_destination=%2Fplugins%2Fservlet%2Fsurveyplugin%2Fadmin
http://localhost:8090/login.action?os_destination=http%3A%2F%2Flocalhost%3A8090%2Fplugins%2Fservlet%2Fsurveyplugin%2Fadminhttp://localhost:8090/login.action?os_destination=%2Fpages%2Fviewpage.action%3Ftitle%3DYet%2Banother%2BSurvey%2Btest%26spaceKey%3DSRV&permissionViolation=true

...the links would take me to the login screen. I enter my credentials and click 'Log In'. At that stage the page goes blank and it appears that it is re-directing me to the survey plugin admin page however I never see the page. After ten seconds of a blank screen I receive the 'This webpage has a redirect loop' error (that's in Chrome, in IE nothing appears, it just keeps loading on the blank page).

I cannot replicate this issue with any other plugin in our Confluence environment.

I hope that has cleared up any misunderstanding.

Regards, David

**Comment by drohne1673** [comment]
the only clear thing is that it is not clear. That is the default implementation of the tutorial, at least it was when it was added. I will check if i can find some source code examples on working plugins...
